sys/kern/kern_intr.c

```diff
@@ -810,8 +810,15 @@ intr_handler_barrier(struct intr_handler *handler)
 		return;
 	}
 	if ((handler->ih_flags & IH_CHANGED) == 0) {
+#ifdef HWPMC_HOOKS
+		struct trapframe *oldframe = curthread->td_intr_frame;
+		curthread->td_intr_frame = NULL;
+#endif
 		handler->ih_flags |= IH_CHANGED;
-		intr_event_schedule_thread(ie, NULL);
+		intr_event_schedule_thread(ie);
+#ifdef HWPMC_HOOKS
+		curthread->td_intr_frame = oldframe;
+#endif
```

Comment:
I'm unsure here.  This keeps functionality unchanged.  Issue is most other points, passing `NULL` as a `struct trapframe *` was due to the author not knowing what to do.  Could be `curthread->td_intr_frame` should simply be left alone here, but I'm unsure.